### PR TITLE
update uint32_t to int32_t

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/2286

### CODE REVIEW

**Review by RickWinter — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-common/inc/azure/keyvault/common/internal/single_page.hpp [hunk 1]
```diff
@@ -17,6 +17,6 @@ namespace Azure { namespace Security { namespace KeyVault { namespace _internal
   {
     virtual ~GetPageResultOptions() = default;
     Azure::Nullable<std::string> NextPageToken;
-    Azure::Nullable<uint32_t> MaxPageResults;
+    Azure::Nullable<int32_t> MaxPageResults;
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-common/inc/azure/keyvault/common/internal/single_page.hpp:20]: Did you check if there were any other places that need fixing? I think I remember seeing a couple when I checked last.

File: sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 ### Breaking Changes
 
+- Updated `MaxPageResults` type to `int32_t` affecting:
+  - GetDeletedKeysOptions.
+  - GetPropertiesOfKeysOptions.
+  - GetPropertiesOfKeyVersionsOptions.
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md:10]:
```suggestion
  - GetDeletedKeysOptions().
  - GetPropertiesOfKeysOptions().
  - GetPropertiesOfKeyVersionsOptions().
```

File: sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,10 @@
 
 ### Breaking Changes
 
+- Updated `MaxPageResults` type to `int32_t` affecting:
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md:7]:
```suggestion
- Updated `MaxPageResults` type to `int32_t`, from `uint32_t`, affecting:
```

**Review by ahsonkhan — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-common/inc/azure/keyvault/common/internal/single_page.hpp (hunk 1 above)
Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-common/inc/azure/keyvault/common/internal/single_page.hpp:20]: I think the others already got fixed :)

File: sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md (hunk 2 above)
Comment (antkmsft) [reply sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md:10]: Suggestion to suggestion: backtick as `GetDeletedKeysOptions()` and so on.

Comment (antkmsft) [reply sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md:10]: @ahsonkhan, I updated your suggestion with backticks, if you don't mind.

Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md:10]: I don't mind at all, but looks like it wasn't applied :p

**Pull request merged**